src/charm.py

```diff
@@ -1146,23 +1167,279 @@ def _addon_reconciliation(self, addon_type):
+        activate_result = self._perform_plugin_activate_or_deactivate(plugin, "activate")
+        if not activate_result.success:
+            return activate_result
+        for option, value in options.items():
+            if isinstance(value, dict):
+                option_update_result = self._wp_option_update(
+                    option=option, value=json.dumps(value), format="json"
+                )
+            else:
+                option_update_result = self._wp_option_update(option=option, value=value)
+            if not option_update_result.success:
+                return self._ExecResult(
+                    success=False,
+                    result=None,
+                    message=f"failed to update option {option} after activating plugin {plugin}"
+                )
+        return self._ExecResult(success=True, result=None, message="")
+
+    def _deactivate_plugin(self, plugin, options):
+        """Deactivate a WordPress plugin and delete WordPress options after deactivation"""
+        deactivate_result = self._perform_plugin_activate_or_deactivate(plugin, "deactivate")
+        if not deactivate_result.success:
+            return deactivate_result
+        for option in options:
+            option_update_result = self._wp_option_delete(option)
+            if not option_update_result.success:
+                return self._ExecResult(
+                    success=False,
+                    result=None,
+                    message=f"failed to delete option {option} after deactivating plugin {plugin}"
+                )
+        return self._ExecResult(success=True, result=None, message="")
+
+    def _plugin_akismet_reconciliation(self):
+        """Reconciliation process for the akismet plugin"""
+        akismet_key = self.model.config["wp_plugin_akismet_key"].strip()
+        if not akismet_key:
+            result = self._deactivate_plugin("akismet", [
+                "akismet_strictness",
+                "akismet_show_user_comments_approved",
+                "wordpress_api_key"
+            ])
+        else:
+            result = self._activate_plugin("akismet", {
+                "akismet_strictness": "0",
+                "akismet_show_user_comments_approved": "0",
+                "wordpress_api_key": akismet_key
+            })
+        if not result.success:
+            raise exceptions.WordPressBlockedStatusException(
+                f"Unable to config akismet plugin, {result.message}"
+            )
+
+    @staticmethod
+    def _encode_openid_team_map(team_map):
+        """Convert wp_plugin_openid_team_map setting to WordPress openid_teams_trust_list option
+
+        example input: site-sysadmins=administrator,site-editors=editor,site-executives=editor
+        """
+        team_map_lines = []
+        i = 0
+        team_map_lines.append("a:{}:{{".format(len(team_map.split(","))))
+        for mapping in team_map.split(","):
+            i = i + 1
```

Comment:
You can automatically get indexes if you wrap an iterable in the builtn `enumerate` function which returns the index and item as a tuple: https://docs.python.org/3/library/functions.html#enumerate

Reply:
Yes, this is copied from the current stable version directly, this function is actually not quite right (encoding a PHP array in Python), I will replace this method in the future.